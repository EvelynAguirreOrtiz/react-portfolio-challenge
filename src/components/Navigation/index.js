import React, { useState } from "react";

function Navigation() {
	return (
		<nav>
			<ul className="flex-row">
				<li className="nav-margin">
					<a data-testid="about" href="#about">
						About Me
					</a>
				</li>
				<li className="nav-margin">
					<a data-testid="portfolio" href="#portfolio">
						Portfolio
					</a>
				</li>
				<li className="nav-margin">
					<a data-testid="resume" href="#resume">
						Resume
					</a>
				</li>
				<li>
					<span>Contact</span>
				</li>
			</ul>
		</nav>
	);
}

export default Navigation;
