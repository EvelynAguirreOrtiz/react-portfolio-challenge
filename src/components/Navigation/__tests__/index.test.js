import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Navigation from "..";

afterEach(cleanup);

describe("Navigation component", () => {
	it("renders", () => {
		render(<Navigation />);
	});
	it("matches snapshot", () => {
		const { asFragment } = render(<Navigation />);
		expect(asFragment()).toMatchSnapshot();
	});
});

describe("links are visible", () => {
	it("inserts text into the links", () => {
		const { getByTestId } = render(<Navigation />);
		expect(getByTestId("about")).toHaveTextContent("About Me");
		expect(getByTestId("portfolio")).toHaveTextContent("Portfolio");
		expect(getByTestId("resume")).toHaveTextContent("Resume");
	});
});
