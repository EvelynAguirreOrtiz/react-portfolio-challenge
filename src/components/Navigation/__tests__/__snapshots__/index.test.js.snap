// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component matches snapshot 1`] = `
<DocumentFragment>
  <nav>
    <ul
      class="flex-row"
    >
      <li
        class="nav-margin"
      >
        <a
          data-testid="about"
          href="#about"
        >
          About Me
        </a>
      </li>
      <li
        class="nav-margin"
      >
        <a
          data-testid="portfolio"
          href="#portfolio"
        >
          Portfolio
        </a>
      </li>
      <li
        class="nav-margin"
      >
        <a
          data-testid="resume"
          href="#resume"
        >
          Resume
        </a>
      </li>
      <li>
        <span>
          Contact
        </span>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
